### PR TITLE
MHV-42069: Health conditions endpoints complete

### PR DESCRIPTION
**Pull request by mattwrightva**
## Summary
The MHV medical records health conditions endpoints are complete. This includes an endpoint for getting all health conditions for a patient and getting a single health condition by its ID. 

## Related issue(s)
https://jira.devops.va.gov/browse/MHV-42069

## Testing done
Tests have been completed for both endpoints utilizing the hapi fhir testing server/database. 

## Screenshots
No UI changes made. 

## What areas of the site does it impact?
This impacts the MHV medical records API. 

## Acceptance criteria
The Health Conditions page displays data from the API
- [ ]  The Chem/hem page displays data from the API
- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

**Review by mmoyer-va — APPROVED** [review]

File: lib/medical_records/client.rb [hunk 1]
```diff
@@ -10,6 +10,18 @@ module MedicalRecords
   # Core class responsible for Medical Records API interface operations
   #
   class Client < Common::Client::Base
+    # loinc codes for clinical notes
+    PHYSICIAN_PROCEDURE_NOTE = '11505-5' # Physician procedure note
```
Comment (acrollet) [inline lib/medical_records/client.rb:14]: 🎉

File: modules/my_health/app/controllers/my_health/v1/conditions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        resource = client.list_conditions(patient_id)
+        raise Common::Exceptions::InternalServerError if resource.blank?
+
+        render json: resource.to_json
+      end
+
+      def show
+        condition_id = params[:id].try(:to_i)
+        resource = client.get_condition(condition_id)
+        raise Common::Exceptions::InternalServerError if resource.blank?
```
Comment (rmtolmach) [inline modules/my_health/app/controllers/my_health/v1/conditions_controller.rb:17]: I see you have some tests checking the successful response. Can you also test the error paths?

Comment (mattwrightva) [reply modules/my_health/app/controllers/my_health/v1/conditions_controller.rb:17]: I added a couple more tests for error paths. Let me know if you think anything else is needed. Thanks!

**Review by rmtolmach — APPROVED** [review]
Looks good!

**Pull request merged**